source/models/table.py

```diff
@@ -47,22 +53,65 @@ def set(self, row):
             old_row.delete()
         row_obj.serialize()
 
-    
-    def get(self):
-        pass
-    
+
+    def get_similar_primarykeys(self, query):
+        primarykeys = []
+        for index_name in query:
+            if index_name in self.table_metadata.index_keys:
+                primarykeys = Index(self, index_name = index_name, index_value = query[index_name]).get_primary_key(index_value = query[index_name])
+                break
+        return primarykeys
+ 
+
+    def get(self, query):
+        self.__colomns_name_validate(query)
+        primarykeys = []
+        if Keys.PRIMARY_KEY in query : # if primarykey in the query
+            primarykeys.append(query[Keys.PRIMARY_KEY])     
+        else:
+            primarykeys = self.get_similar_primarykeys(query = query)
+            if not len(primarykeys):# compare all data in the table with the query
```

Comment:
You need 2 information to descide that you need to loop for all data. 
- index[] is empty 
- query doesn't have in key indexable. 

so I suggest `get_similar_primarykeys` return NULL if there is no key in the query indexed.

Reply:
yes , that i made

Reply:
i prefere to return an empty list as primarykeys variable in get function is list,
so if I return null I should change it.
and I think null and empty list get the same menning

Reply:
if you want me to change it also, I will do it